src/app/ReadHandler.cpp

```diff
@@ -293,7 +305,7 @@ bool ReadHandler::CheckEventClean(EventManagement & aEventManager)
     if (mCurrentPriority == PriorityLevel::Invalid)
     {
         // Upload is not in middle, previous mLastScheduledEventNumber is not valid, Check for new events, and set a checkpoint
-        for (size_t index = 0; index < ArraySize(mSelfProcessedEvents); index++)
+        for (int index = ArraySize(mSelfProcessedEvents); index >= 0; index--)
```

Comment:
previous loop has ArraySize counts. This one seems ArraySize + 1 (because we start with size and include 0 counding down). Is this correct or off-by-one? It feels off-by one to me.

Reply:
yep, thanks for catching up it. just fix it.